File: backend/src/main/java/com/campus/Campus_Connect/features/honor/dto/response/MedalsResponse.java
package com.campus.Campus_Connect.features.honor.dto.response;

import lombok.*;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MedalsResponse {

    private MedalResponse gold;

    private MedalResponse silver;

    private MedalResponse bronze;

    private Integer awardedCount;

}

File: android-app/app/src/main/java/com/example/campusconnect/feature/events/data/remote/response/MedalsResponse.kt
package com.example.campusconnect.feature.events.data.remote.response

import com.google.gson.annotations.SerializedName

data class MedalsResponse(

    @SerializedName("gold")
    val gold: MedalResponse,

    @SerializedName("silver")
    val silver: MedalResponse,

    @SerializedName("bronze")
    val bronze: MedalResponse,

    @SerializedName("awardedCount")
    val awardedCount: Int
)